### PR TITLE
Add query coord watch delta channel

### PR DESCRIPTION
**Pull request by godchen0212**
Add query coord watch delta channel
Issue #11217

Signed-off-by: godchen <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/querycoord/task.go [hunk 1]
```diff
@@ -1483,13 +1603,29 @@ func (ht *handoffTask) execute(ctx context.Context) error {
 					}
 				}
 			}
+			// init delta channels for sealed segments.
+			if loadSegmentReq != nil && len(watchDeltaChannelReqs) != len(recoveryInfo.Channels) {
+				for _, info := range recoveryInfo.Channels {
+					deltaChannel := info.ChannelName
+					deltaInfo := proto.Clone(info).(*datapb.VchannelInfo)
```
Comment (xige-16) [inline internal/querycoord/task.go:1610]: dataInfo created but not used

File: internal/querycoord/task.go [hunk 2]
```diff
@@ -1483,13 +1603,29 @@ func (ht *handoffTask) execute(ctx context.Context) error {
 					}
 				}
 			}
+			// init delta channels for sealed segments.
+			if loadSegmentReq != nil && len(watchDeltaChannelReqs) != len(recoveryInfo.Channels) {
+				for _, info := range recoveryInfo.Channels {
```
Comment (xige-16) [inline internal/querycoord/task.go:1608]: does dmChannels  equal to deltaCahnnels?

File: internal/querycoord/task_scheduler.go [hunk 3]
```diff
@@ -463,6 +463,8 @@ func (scheduler *TaskScheduler) processTask(t task) error {
 				protoSize = proto.Size(childTask.(*loadSegmentTask).LoadSegmentsRequest)
 			case commonpb.MsgType_WatchDmChannels:
 				protoSize = proto.Size(childTask.(*watchDmChannelTask).WatchDmChannelsRequest)
+			case commonpb.MsgType_WatchDeltaChannels:
+				protoSize = proto.Size(childTask.(*watchDeltaChannelTask).WatchDeltaChannelsRequest)
```
Comment (xige-16) [inline internal/querycoord/task_scheduler.go:467]: reloadFromKV and wait waitActivateTaskDone function should deal WatchDeltaChannelsRequest

**Comment by xige-16** [comment]
How to ensure that no less deltelte messages will be consumed after query node is down

**Comment by godchen0212** [comment]
The form of deltaChannel and dmlChannel are currently the same, so VchannelInfo is used. In the future, we will use seek position to record delta channel consumption records to ensure that querynode will not consume too much data or omit relevant data.

File: internal/querycoord/task_scheduler.go [hunk 4]
```diff
@@ -816,6 +833,8 @@ func (scheduler *TaskScheduler) waitActivateTaskDone(wg *sync.WaitGroup, t task,
 			redoFunc1()
 		case commonpb.MsgType_WatchDmChannels:
 			redoFunc1()
+		case commonpb.MsgType_WatchDeltaChannels:
+			redoFunc1()
```
Comment (xige-16) [inline internal/querycoord/task_scheduler.go:837]: should use redoFunc1, reschedule watchDeleteChannel is Pointless

File: internal/querycoord/task.go [hunk 5]
```diff
@@ -1974,6 +2139,23 @@ func assignInternalTask(ctx context.Context,
...
 		}
+
+		for _, req := range watchDeltaChannelRequests {
+			ctx = opentracing.ContextWithSpan(context.Background(), sp)
+			req.NodeID = nodeID
+			baseTask := newBaseTask(ctx, parentTask.getTriggerCondition())
+			baseTask.setParentTask(parentTask)
+			watchDeltaTask := &watchDeltaChannelTask{
+				baseTask:                  baseTask,
+				WatchDeltaChannelsRequest: req,
```
Comment (xige-16) [inline internal/querycoord/task.go:2150]: should deep copy watchDeltaChannelRequest

**Comment by xige-16** [comment]
lgtm

**Comment by xige-16** [comment]
seek position of delta channel should be save in segment checkout point

**Comment by xige-16** [comment]
loadsegment reschedule should be refactored to genenrate watchDeltaChannelRequests

**Pull request merged**